From 9af66319e0ef3e6f4308d2bf4b4ee7a944a053e0 Mon Sep 17 00:00:00 2001
From: Achill Gilgenast <fossdd@pwned.life>
Date: Sat, 26 Jul 2025 20:27:24 +0200
Subject: [PATCH 7/7] FROMLIST: libbpf: avoid possible use of uninitialized
 mod_len

If not fn_name, mod_len does never get initialized which fails now with
gcc15 on Alpine Linux edge:

	libbpf.c: In function 'find_kernel_btf_id.constprop':
	libbpf.c:10100:33: error: 'mod_len' may be used uninitialized [-Werror=maybe-uninitialized]
	10100 |                 if (mod_name && strncmp(mod->name, mod_name, mod_len) != 0)
	      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
	libbpf.c:10070:21: note: 'mod_len' was declared here
	10070 |         int ret, i, mod_len;
	      |                     ^~~~~~~

Link: https://lore.kernel.org/lkml/20250729094611.2065713-1-fossdd@pwned.life/
Fixes: 8f8a024272f3 ("libbpf: support "module: Function" syntax for tracing programs")
Signed-off-by: Achill Gilgenast <fossdd@pwned.life>
---
 tools/lib/bpf/libbpf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/lib/bpf/libbpf.c b/tools/lib/bpf/libbpf.c
index c8e29c52d28c..cdef24c31d11 100644
--- a/tools/lib/bpf/libbpf.c
+++ b/tools/lib/bpf/libbpf.c
@@ -10067,7 +10067,7 @@ static int find_kernel_btf_id(struct bpf_object *obj, const char *attach_name,
 			      enum bpf_attach_type attach_type,
 			      int *btf_obj_fd, int *btf_type_id)
 {
-	int ret, i, mod_len;
+	int ret, i, mod_len = 0;
 	const char *fn_name, *mod_name = NULL;
 
 	fn_name = strchr(attach_name, ':');
-- 
2.50.1

